library/Core.cpp

```diff
@@ -1733,7 +1723,9 @@ void Core::Suspend()
 
 void Core::Resume()
 {
+#ifndef NDEBUG
     auto tid = this_thread::get_id();
+#endif
     lock_guard<mutex> lock(d->AccessMutex);
 
     assert(d->df_suspend_depth > 0 && d->df_suspend_thread == tid);
```

Comment:
`assert(d->df_suspend_depth > 0 && d->df_suspend_thread == this_thread::get_id());` might be cleaner than the NDEBUG guard above (also around line 1767).